app/code/Magento/Inventory/Model/ReservationAppend.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Model;
+
+use Magento\Framework\Exception\CouldNotSaveException;
+use Magento\Framework\Exception\InputException;
+use Magento\Inventory\Model\ResourceModel\Reservation\SaveMultiple;
+use Magento\InventoryApi\Api\Data\ReservationInterface;
+use Magento\InventoryApi\Api\ReservationAppendInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @inheritdoc
+ */
+class ReservationAppend implements ReservationAppendInterface
+{
+    /**
+     * @param SaveMultiple $saveMultiple
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        SaveMultiple $saveMultiple,
+        LoggerInterface $logger
+    ) {
+        $this->saveMultiple = $saveMultiple;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(array $reservations)
+    {
+        /** @var ReservationInterface $reservation */
+        foreach ($reservations as $reservation) {
+            if (!is_null($reservation->getReservationId())) {
```

Comment:
@naydav ReservationAppend service is used to add newly created reservations.
Here we @aleron75  added a check which prevents that someone retrieves a reservation, change some data, and will try to save again using ReservationAppend service.

Current code considers that Database Generated Ids are used, so that entity which has an ReservationId considered as already created, thus **can't be appended one more time**.

So, the logic is correct as for me